src/widgets/WalletModal/WalletCard.tsx

```diff
@@ -20,10 +21,22 @@ const WalletCard: React.FC<Props> = ({ login, walletConfig, onDismiss, mb }) =>
       height='40px'
       borderBottom='1px solid #0d1b24'
       onClick={() => {
-        if (!window.ethereum && walletConfig.redirectUrl) return;
-        login(walletConfig.connectorId);
-        window.localStorage.setItem(connectorLocalStorageKey, walletConfig.connectorId);
-        onDismiss();
+        // TW point to WC on desktop
+        if (title === "TrustWallet" && walletConnectConfig && isDesktop) {
+          login(walletConnectConfig.connectorId);
+          localStorage?.setItem(walletLocalStorageKey, walletConnectConfig.title);
+          localStorage?.setItem(connectorLocalStorageKey, walletConnectConfig.connectorId);
+          onDismiss();
+          return;
+        }
```

Comment:
I think we can simplify this part, we can remove `line 24` to `line 31`

for `line 32`, we can use this
`if (!window.ethereum && walletConfig.redirectUrl && walletConfig.connectorId !== ConnectorNames.WalletConnect) {`

and in `src/widgets/WalletModal/config.tsx`
we can update TrustWallet Object to this
```
{
    title: "TrustWallet",
    icon: TrustWallet,
    connectorId: isDesktop ? ConnectorNames.WalletConnect :  ConnectorNames.Injected,
    redirectUrl: `https://link.trustwallet.com/open_url?coin_id=20000714&url=https://summitswap.finance/`
  },
  ```

Reply:
This method was copied from pancakeswap, but yea we can improve it